src/App.test.js

```diff
@@ -668,6 +668,46 @@ describe('App component', () => {
       });
     });
 
+    it('check utm parameters', async () => {
+      // Arrange
+      const dependencies = {
+        sessionManager: createDoubleSessionManager(),
+        localStorage: createLocalStorageDouble(),
+        dopplerSitesClient: dopplerSitesClientDouble,
+      };
+
+      createJsonParse();
+
+      // Act
+      act(() => {
+        render(
+          <AppServicesProvider forcedServices={dependencies}>
+            <Router
+              initialEntries={[
+                '/signup?origin=testOrigin&utm_source=test&utm_campaign=testcampaign&utm_medium=testmedium&utm_term=testterm',
```

Comment:
It would be nice to see the at least two items in cookies history. Perhaps you could assume you have one entrance as @andresmoschini suggested in the daily.

Reply:
update with this test